modules/core/shared/src/main/scala/eu/timepit/refined/types/string.scala

```diff
@@ -17,6 +17,11 @@ object string {
   type TrimmedString = String Refined MatchesRegex[W.`"""^(?!\\s).*(?<!\\s)"""`.T]
 
   object TrimmedString extends RefinedTypeOps[TrimmedString, String]
+
+  /** A `String` representing a hexadecimal number */
+  type HexStringSpec = MatchesRegex[W.`"""[0-9a-f]+"""`.T]
```

Comment:
@NeQuissimus Should we also allow upper case letters `A-F`? What about an optional `0x` prefix? I think that a prefix would be fine for `HexString` but may be inappropriate for the message digests types. Maybe `HexStringSpec` should be private to `digests` to sidestep these potential conflicts?

Reply:
The prefix is not really much more than an artifact to identify a hexadecimal number to humans or when the information of it being a hex number is not available. I think that is exactly the information we are adding with the refined type.
I will make the change to allow for upper-case letters (but not mixed-upper/lower).